src/cgame/default/cg_entity.c

```diff
@@ -26,6 +26,9 @@
  */
 _Bool Cg_IsSelf(const cl_entity_t *ent) {
 
+	if (ent->current.number == cgi.client->client_num + 1)
```

Comment:
This is a more direct way to check this, anyway, and it works no matter what model (e.g. giblet) our client has.